### DISCUSSION TITLE
carla-single: How to get Jack client name of plugin automatically?

### DISCUSSION THREAD

**Discussion by madskjeldgaard** [falkTX/Carla#1429] [Q&A]
Hello all
I recently discovered the awesome `carla-single` utility. I have a question: How can I get (or maybe set) the name of a plugin's jack client? I need this to fully automate some stuff. For example, running `FabFilter Pro-C2` becomes `Pro-C2` in the jack client. This is a bit difficult to predict. I thought that the label option would help with this but labels don't seem to change this name (and I probably misunderstood how it's supposed to be used?) Thanks!

**Comment by falkTX** [comment]
label is for identifying sub-plugins, like those in ladspa and dssi.
what you need is `CARLA_CLIENT_NAME` env var.

**Comment by madskjeldgaard** [comment]
Fantastic, this works perfectly. Thnak you @falkTX :

```bash
CARLA_CLIENT_NAME="FunkyDonkey" carla-single vst3 .vst3/yabridge/FabFilter\ Pro-C\ 2.vst3
```
